ibis/backends/tests/test_generic.py

```diff
@@ -1180,3 +1188,92 @@ def test_hash_consistent(backend, alltypes):
     h2 = alltypes.string_col.hash().execute(limit=10)
     tm.assert_series_equal(h1, h2)
     assert h1.dtype in ("i8", "uint64")  # polars likes returning uint64 for this
+
+
+@pytest.mark.notimpl(
+    [
+        "pandas",
+        "dask",
+        "bigquery",
+        "datafusion",
+        "druid",
+        "impala",
+        "mssql",
+        "mysql",
+        "oracle",
+        "postgres",
+        "pyspark",
+        "snowflake",
+        "sqlite",
+    ]
+)
+@pytest.mark.parametrize(
+    ("from_val", "to_type", "expected"),
+    [
+        param(0, "float", 0.0),
+        param(0.0, "int", 0),
+        param("0", "int", 0),
+        param("0.0", "float", 0.0),
+        param(
+            "a",
+            "int",
+            None,
+            marks=pytest.mark.notyet(["polars"], reason="polars casts to nan"),
+        ),
+        param(
+            datetime.datetime.now(),
+            "int",
+            None,
+            marks=pytest.mark.notyet(
+                ["clickhouse", "trino", "polars"],
+                reason=[
+                    "clickhouse casts this to 1686156304",
+                    "trino raises error",
+                    "polars casts this to 1686157562300325000",
+                ],
+            ),
+        ),
+    ],
+)
+def test_try_cast_expected(con, from_val, to_type, expected):
+    assert con.execute(ibis.literal(from_val).try_cast(to_type)) == expected
+
+
+@pytest.mark.notimpl(
+    [
+        "pandas",
+        "dask",
+        "bigquery",
+        "datafusion",
+        "druid",
+        "impala",
+        "mssql",
+        "mysql",
+        "oracle",
+        "postgres",
+        "pyspark",
+        "snowflake",
+        "sqlite",
+    ]
+)
+@pytest.mark.parametrize(
+    ("from_val", "to_type", "func"),
+    [
+        param("a", "float", np.isnan),
+        param(
+            datetime.datetime.now(),
+            "float",
+            np.isnan,
+            marks=pytest.mark.notyet(
+                ["clickhouse", "trino", "polars"],
+                reason=[
+                    "clickhouse casts this to 1686156304",
+                    "trino raises TrinoUserError",
+                    "polars casts this to 1686157562300325000",
+                ],
+            ),
```

Comment:
the `notyet` (and `xfail`) marker expects a string for the `reason` -- if you want to give individual reasons, you need to add them as separate marks, so something like:
```suggestion
            marks=[pytest.mark.notyet(
                ["clickhouse"], reason="clickhouse casts this to 1686156304"),
                pytest.mark.notyet(["trino"], reason="trino raises TrinoUserError"),
                pytest.mark.notyet(["polars"], reason="polars casts this to 1686157562300325000"),
                ],
                ],
            ),
```

(I might've missed a comma in there, or have too many brackets, but the general idea is to pass a list of individual `notyet`s to `marks`)